### ISSUE TITLE
Consider supporting PostgreSQL/openGauss search path feature

### ISSUE THREAD

**Issue by strongduanmu** [apache/shardingsphere#17204]
### Is your feature request related to a problem?

No.

### Describe the feature you would like.

After issue #14004 is completed, ShardingSphere has supported SQL such as `CREATE SCHEMA`, `ALTER SCHEMA` and `DROP SCHEMA`, and allows users to specify schema modifiers when executing SQL, such as:

```sql
SELECT * FROM test.t_order;
```

This SQL will query the data of the t_order table from the test schema.

In order to improve the experience of using PostgreSQL/openGauss schema, we need to support the search_path parameter. Through the `SET SEARCH_PATH TO test, public;` statement, users can dynamically switch the schema without going through schema.t_order. For more detail about search_path, we can refer this doc——https://www.postgresql.org/docs/14/ddl-schemas.html#DDL-SCHEMAS-PATH.

**Comment by zjcnb** [comment]
@strongduanmu Hi,i want to do this. Can you assign me? Thx

**Comment by strongduanmu** [comment]
@zjcnb Of course. We can consider implementing the global search path first by adding a configuration.

**Comment by zhaojinchao95** [comment]
Hi, i am sort out what needs to be done. @FlyingZC Can you do it with me?

- [ ] Add search path configuation for server.yaml props
- [ ] Refactor metadata structure
- [ ] Check all schemas are consistent
- [ ] Set search path to jdbc url
- [ ] Refactor get schema logic
- [ ] Clear up session search path

**Comment by FlyingZC** [comment]
> Hi, i am sort out what needs to be done. @FlyingZC Can you do it with me?
> 
> * [ ]  Add search path configuation for server.yaml props
> * [ ]  Refactor metadata structure
> * [ ]  Check all schemas are consistent
> * [ ]  Set search path to jdbc url
> * [ ]  Refactor get schema logic
> * [ ]  Clear up session search path

Ok,I'd like to try that.

**Comment by strongduanmu** [comment]
This issue is currently low priority, so I will remove milestone.

**Comment by Y-sir** [comment]
@zhaojinchao95 @FlyingZC I would like to ask this question, can it be completed in the next version?

**Comment by bolyage** [comment]
any update？

**Comment by huie2018** [comment]
Why has it been two years, and it is still not supported

**Comment by strongduanmu** [comment]
> Why has it been two years, and it is still not supported

Apparently, no volunteers are interested in implementing this feature. If you are interested in implementing it, feel free to claim it and submit a PR.

**Comment by mlmw92** [comment]
I found that adding the following code to `PostgreSQLDatabaseMetaData` can solve this problem

```java
    @Override
    public Optional<String> getDefaultSchema() {
        return this.defaultSchema != null ? Optional.of(this.defaultSchema) : Optional.of("public");
    }

    @Override
    public String getSchema(Connection connection) {
        this.defaultSchema = DialectDatabaseMetaData.super.getSchema(connection);
        return this.defaultSchema;
    }
```

**Comment by Liuxing778** [comment]
Hi, is the issue be solved now?I need to use springboot 3.4.1 with shardingshpere 5.5.2. and the database is postgresql, it is blocked by this issue

**Comment by strongduanmu** [comment]
> Hi, is the issue be solved now?I need to use springboot 3.4.1 with shardingshpere 5.5.2. and the database is postgresql, it is blocked by this issue

@Liuxing778 At the moment, this task is not completed, are you interested in completing it?

**Comment by LoveYou3000** [comment]
> getSchema

I think this solution is not an ideal one because adding an attribute to `PostgreSQLDatabaseMetaData` would make it a stateful class. If an instance of this class does not call the `getSchema` method but directly calls the `getDefaultSchema` method, it will not be able to obtain the schema in the connection; instead, it will obtain the default `public` schema.
